FiveCalls/FiveCalls/MyImpactViewController.swift

```diff
@@ -103,14 +104,19 @@ class MyImpactViewController : UITableViewController {
     
     func displayStats() {
         viewModel = ImpactViewModel(logs: ContactLogs.load(), stats: userStats)
-        let weeklyStreakCount = viewModel.weeklyStreakCount
+
+        // use the server weekly streak if we have it
+        let weeklyStreakCount = self.weeklyStreak ?? viewModel.weeklyStreakCount
         streakLabel.text = weeklyStreakMessage(for: weeklyStreakCount)
```

Comment:
all of this should be in `ImpactViewModel`, including this `weeklyStreakMessage`. The result should just be 

```swift
streakLabel.text = viewModel.streakMessage
```

Reply:
👍